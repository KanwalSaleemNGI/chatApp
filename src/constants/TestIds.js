export default {
  authButton: {
    title: 'title',
    onPress: 'onPress',
  },
  chatInfo: {
    onPress: 'onPress',
    image: 'image',
    name: 'name',
    messageText: 'userChatData',
    icon: 'icon',
    photoText: 'photoText',
    date: 'date',
  },
  login: {
    emailInput: 'emailInput',
    passwordInput: 'passwordInput',
  },
};
