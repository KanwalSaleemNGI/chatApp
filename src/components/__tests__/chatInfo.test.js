import React from 'react';
import {ChatInfo} from '../';
import {fireEvent, render} from '@testing-library/react-native';
import TestIds from '../../constants/TestIds';
import {NavigationContainer} from '@react-navigation/native';

describe('chatInfo comp', () => {
  const onPress = jest.fn();
  const imageUrl =
    'https://media.istockphoto.com/vectors/user-icon-flat-isolated-on-white-background-user-symbol-vector-vector-id1300845620?k=20&m=1300845620&s=612x612&w=0&h=f4XTZDAv7NPuZbG0habSpU0sNgECM0X7nbKzTUta3n8=';
  const firstName = 'Kanwal';
  const lastName = 'saleem';
  const message = 'Hi';
  const date = '10/10/2022';

  const renderHandler = () => {
    return render(
      <ChatInfo
        item={{
          recentChat: {text: message, createdDate: date},
          userChatData: {
            userImg: imageUrl,
            firstName,
            lastName,
          },
        }}
      />,
    );
  };

  it('should match snapshot', () => {
    //arrange
    const {toJSON} = renderHandler();

    //act

    //assert
    expect(toJSON()).toMatchSnapshot();
  });

  it('should properly render the image & name', () => {
    //arrange
    const {getByTestId, queryByTestId} = renderHandler();

    //act

    //assert
    expect(queryByTestId(TestIds.chatInfo.image)).toBeTruthy();
    expect(getByTestId(TestIds.chatInfo.name).props.children).toEqual(
      `${firstName} ${lastName}`,
    );
  });

  it('should properly renders mesaage', () => {
    const {getByTestId} = renderHandler();

    expect(getByTestId(TestIds.chatInfo.messageText).props.children).toEqual(
      message,
    );
  });
  it('should properly renders date', () => {
    const {getByTestId} = renderHandler();

    expect(getByTestId(TestIds.chatInfo.date).props.children).toEqual(date);
  });

  it('should properly press the button', () => {
    const {getByTestId} = renderHandler();
    fireEvent(getByTestId(TestIds.chatInfo.onPress), 'press');
    // expect(onPress).toHaveBeenCalled();
  });
});
