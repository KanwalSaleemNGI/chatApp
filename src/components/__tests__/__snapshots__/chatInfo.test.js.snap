// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInfo comp should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F6F6F6",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="onPress"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "20%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://media.istockphoto.com/vectors/user-icon-flat-isolated-on-white-background-user-symbol-vector-vector-id1300845620?k=20&m=1300845620&s=612x612&w=0&h=f4XTZDAv7NPuZbG0habSpU0sNgECM0X7nbKzTUta3n8=",
          }
        }
        style={
          Object {
            "borderRadius": 25,
            "height": 50,
            "width": 50,
          }
        }
        testID="image"
      />
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 5,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Roboto-Medium",
            "fontSize": 16,
          }
        }
        testID="name"
      >
        Kanwal saleem
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#655A58",
            "fontFamily": "Roboto-Regular",
            "fontSize": 14,
          }
        }
        testID="userChatData"
      >
        Hi
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#655A58",
          "fontFamily": "Roboto-Regular",
          "fontSize": 12,
        }
      }
      testID="date"
    >
      10/10/2022
    </Text>
  </View>
</View>
`;
